astropy/constants/constant.py

```diff
@@ -118,9 +118,9 @@ def __new__(cls, abbrev, name, value, unit, uncertainty,
         inst = np.array(value).view(cls)
 
         if system in instances:
-                warnings.warn('Constant {!r} already has a definition in the '
-                              '{!r} system from {!r} reference'.format(
-                              name, system, reference), AstropyUserWarning)
```

Comment:
Optional, but now is as good a time as any to upgrade this to f-strings.

Reply:
I wonder why pyupgrade didn't do that.

Reply:
Possibly related to this: https://github.com/python/cpython/issues/90358

Reply:
I see the potential problem, but I don't think it applies in this case since ``.format`` is used before ``warnings.warn`` sees the string. I don't think there's delayed eval here

Reply:
Our use of f-strings actually seems safe - that issue mentions a very particular problem.

Reply:
The issue mentioned above is specific to `logging` which does internally format strings, so not the case here.